engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/event/IntermediateThrowEventProcessor.java

```diff
@@ -40,43 +57,152 @@ public Class<ExecutableIntermediateThrowEvent> getType() {
+      if (element.getJobWorkerProperties() != null) {
+        variableMappingBehavior
+            .applyInputMappings(activating, element)
+            .flatMap(ok -> jobBehavior.createNewJob(activating, element))
+            .ifRightOrLeft(
+                ok -> stateTransitionBehavior.transitionToActivated(activating),
+                failure -> incidentBehavior.createIncident(failure, activating));
+      }
+    }
+
+    @Override
+    public void onComplete(
+        final ExecutableIntermediateThrowEvent element, final BpmnElementContext completing) {
+      variableMappingBehavior
+          .applyOutputMappings(completing, element)
+          .flatMap(ok -> stateTransitionBehavior.transitionToCompleted(element, completing))
+          .ifRightOrLeft(
+              completed -> stateTransitionBehavior.takeOutgoingSequenceFlows(element, completed),
+              failure -> incidentBehavior.createIncident(failure, completing));
+    }
+
+    @Override
+    public void onTerminate(
+        final ExecutableIntermediateThrowEvent element, final BpmnElementContext terminating) {
+      if (element.getJobWorkerProperties() != null) {
+        jobBehavior.cancelJob(terminating);
+      }
+    }
+  }
+
+  private class LinkIntermediateThrowEventBehavior implements IntermediateThrowEventBehavior {
+    @Override
+    public boolean isSuitableForEvent(final ExecutableIntermediateThrowEvent element) {
+      return element.isLinkThrowEvent();
+    }
+
+    @Override
+    public void onActivate(
+        final ExecutableIntermediateThrowEvent element, final BpmnElementContext activating) {
+      final var link = element.getLink();
+      ensureNotNull("link", link);
+
+      final var name = link.getName();
+      ensureNotNullOrEmpty("name", name);
+
+      eventPublicationBehavior
+          .findLinkCatchEvent(link.getName(), activating)
+          .ifRightOrLeft(
+              eventTuple -> {
+                final var catchEvent = (ExecutableCatchEventElement) eventTuple.getCatchEvent();
+                final var processInstanceRecord = eventTuple.getElementInstance().getValue();
+                elementActivationBehavior.activateElement(processInstanceRecord, catchEvent);
+
+                final var activated = stateTransitionBehavior.transitionToActivated(activating);
+                stateTransitionBehavior.completeElement(activated);
+              },
+              failure -> incidentBehavior.createIncident(failure, activating));
+    }
+
+    @Override
+    public void onComplete(
+        final ExecutableIntermediateThrowEvent element, final BpmnElementContext completing) {
+      variableMappingBehavior
+          .applyOutputMappings(completing, element)
+          .flatMap(ok -> stateTransitionBehavior.transitionToCompleted(element, completing))
+          .ifRightOrLeft(
+              completed -> stateTransitionBehavior.takeOutgoingSequenceFlows(element, completed),
```

Comment:
> Instead, we should activate the link catch event after the link throw event transitioned to completed

❌ I don't quite agree with this. When throw event completed, the process instance will complete immediately because there is no active element instance. 

 I think we should implement the following code.
 ```java
 public void onComplete(
        final ExecutableIntermediateThrowEvent element, final BpmnElementContext completing) {
      final var link = element.getLink();
      variableMappingBehavior
          .applyOutputMappings(completing, element)
          .flatMap(
              ok -> {
                elementActivationBehavior.activateElement(completing.getRecordValue(), link.getCatchEventElement());
                return stateTransitionBehavior.transitionToCompleted(element, completing);
              })
          .ifLeft(failure -> incidentBehavior.createIncident(failure, completing));
    }
 ```